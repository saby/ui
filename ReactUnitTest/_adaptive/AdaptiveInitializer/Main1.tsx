import Layout1 from './Layout1';
import { AdaptiveInitializerConfig } from 'UICore/Adaptive';

export default function Main1() {
    return (
        <AdaptiveInitializerConfig isPhoneForced={true}>
            <Layout1 />
        </AdaptiveInitializerConfig>
    );
}
