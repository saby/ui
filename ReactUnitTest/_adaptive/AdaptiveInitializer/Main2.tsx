import Layout1 from './Layout1';
import { AdaptiveInitializerConfig } from 'UICore/Adaptive';

export default function Main2() {
    return (
        <AdaptiveInitializerConfig isPhoneForced={false}>
            <Layout1 />
        </AdaptiveInitializerConfig>
    );
}
