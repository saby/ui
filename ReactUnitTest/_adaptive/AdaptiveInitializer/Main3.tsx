import Layout1 from './Layout1';

export default function Main1() {
    return <Layout1 />;
}
