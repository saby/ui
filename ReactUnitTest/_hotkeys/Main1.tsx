import { KeyHook } from 'UI/HotKeys';
import { constants } from 'Env/Env';
import { useCallback, useState } from 'react';
import * as React from 'react';

export default React.memo(function Main1() {
    const [value, setValue] = useState(0);
    const keyDownHandler = useCallback(() => {
        setValue(value + 1);
    }, []);
    return (
        <div onKeyDown={keyDownHandler}>
            <KeyHook defaultActions={[{ keyCode: constants.key.enter }]}>
                <div>обработчик keydown сработал: {value}</div>
            </KeyHook>
        </div>
    );
});
