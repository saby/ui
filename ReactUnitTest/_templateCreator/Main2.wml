<ReactUnitTest._templateCreator.Hoc value="{{value}}">
   <ReactUnitTest._templateCreator.Child value="{{value}}"/>
</ReactUnitTest._templateCreator.Hoc>
