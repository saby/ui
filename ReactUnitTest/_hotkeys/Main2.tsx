import { Component } from 'react';
import Main1 from './Main1';

interface IState {
    className: string;
}

export default class Main2 extends Component<any, IState> {
    constructor(props: any) {
        super(props);
        this.state = { className: '' };
    }

    setClassName(className: string) {
        this.setState({ className });
    }
    render() {
        return (
            <div className={this.state.className}>
                <Main1 />
            </div>
        );
    }
}
