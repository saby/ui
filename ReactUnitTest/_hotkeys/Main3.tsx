import { Component } from 'react';
import Main1 from './Main1';

interface IState {
    name: string;
}

export default class Main2 extends Component<any, IState> {
    constructor(props: any) {
        super(props);
        this.state = { name: '' };
    }

    setContainer(name: string) {
        this.setState({ name });
    }
    render() {
        return this.state.name === 'article' ? (
            <article>
                <Main1 />
            </article>
        ) : (
            <div>
                <Main1 />
            </div>
        );
    }
}
