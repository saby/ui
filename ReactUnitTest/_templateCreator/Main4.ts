import { Control, TemplateFunction } from 'UI/Base';
import * as template from 'wml!ReactUnitTest/_templateCreator/Main4';

export default class Main4 extends Control {
    _template: TemplateFunction = template;
}
