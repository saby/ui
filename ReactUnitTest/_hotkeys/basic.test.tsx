/**
 * @jest-environment jsdom
 */
import { Storage } from 'UICore/_adaptive/Aspects';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import Bootstrap from 'UI/Bootstrap';
import Main1 from './Main1';
import Main2 from './Main2';
import Main3 from './Main3';
import Main4 from './Main4';
import { fireEvent } from '@testing-library/react';
import { createRef } from 'react';

describe('HotKeys', () => {
    let container: HTMLDivElement;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });
    afterEach(() => {
        Storage.getInstance()._clear();
        unmountComponentAtNode(container);
        container.remove();
        document.body.removeAttribute('style');
        Storage.getInstance()._clear();
        jest.restoreAllMocks();
    });

    test('basic', () => {
        act(() => {
            render(
                <Bootstrap>
                    <div className="mainDiv">
                        <div className="unit-test-input"></div>
                        <Main1 />
                    </div>
                </Bootstrap>,
                container
            );
        });
        const input = container.querySelector('.unit-test-input');
        if (input) {
            fireEvent.keyDown(input, {
                which: 13,
            });
        }
        expect(container.querySelector('.mainDiv')).toMatchSnapshot();
    });
    test('родительский контрол перерисовывается', () => {
        const ref = createRef();
        act(() => {
            render(
                <Bootstrap>
                    <div className="mainDiv">
                        <div className="unit-test-input"></div>
                        <Main2 ref={ref} />
                    </div>
                </Bootstrap>,
                container
            );
        });
        ref.current.setClassName('abc');
        const input = container.querySelector('.unit-test-input');
        if (input) {
            fireEvent.keyDown(input, {
                which: 13,
            });
        }
        expect(container.querySelector('.mainDiv')).toMatchSnapshot();
    });

    test('родительский контрол пересоздается', () => {
        const ref = createRef();
        act(() => {
            render(
                <Bootstrap>
                    <div className="mainDiv">
                        <div className="unit-test-input"></div>
                        <Main3 ref={ref} />
                    </div>
                </Bootstrap>,
                container
            );
        });
        ref.current.setContainer('article');
        const input = container.querySelector('.unit-test-input');
        if (input) {
            fireEvent.keyDown(input, {
                which: 13,
            });
        }
        expect(container.querySelector('.mainDiv')).toMatchSnapshot();
    });
    test('комплексный тест - 2 пересоздающихся соседа', () => {
        const ref1 = createRef();
        const ref2 = createRef();
        act(() => {
            render(
                <Bootstrap>
                    <div className="mainDiv">
                        <div className="unit-test-input"></div>
                        <Main3 ref={ref1} />
                        <Main3 ref={ref2} />
                    </div>
                </Bootstrap>,
                container
            );
        });

        ref1.current.setContainer('article');
        ref2.current.setContainer('article');
        ref1.current.setContainer('div');
        ref2.current.setContainer('div');
        // ref2.current.setContainer('article');
        const input = container.querySelector('.unit-test-input');
        fireEvent.keyDown(input, { which: 13 });
        expect(container.querySelector('.mainDiv')).toMatchSnapshot('complex');
    });
    test('комплексный тест - перерисовывающийся и пересоздающийся соседи', () => {
        const ref1 = createRef();
        const ref2 = createRef();
        act(() => {
            render(
                <Bootstrap>
                    <div className="mainDiv">
                        <div className="unit-test-input"></div>
                        <Main2 ref={ref1} />
                        <Main3 ref={ref2} />
                    </div>
                </Bootstrap>,
                container
            );
        });

        ref1.current.setClassName('abc');
        ref2.current.setContainer('article');
        ref1.current.setClassName('abc2');
        ref2.current.setContainer('div');
        // ref2.current.setContainer('article');
        const input = container.querySelector('.unit-test-input');
        fireEvent.keyDown(input, { which: 13 });
        expect(container.querySelector('.mainDiv')).toMatchSnapshot('complex');
    });
    test('комплексный тест - первый сосед пропадает, второй должен перехватить инициативу', () => {
        const ref1 = createRef();
        const ref2 = createRef();
        act(() => {
            render(
                <Bootstrap>
                    <div className="mainDiv">
                        <div className="unit-test-input"></div>
                        <Main4 ref={ref1} />
                        <Main3 ref={ref2} />
                    </div>
                </Bootstrap>,
                container
            );
        });

        ref1.current.setContainer('article');
        const input = container.querySelector('.unit-test-input');
        fireEvent.keyDown(input, { which: 13 });
        expect(container.querySelector('.mainDiv')).toMatchSnapshot('complex');
    });
});
