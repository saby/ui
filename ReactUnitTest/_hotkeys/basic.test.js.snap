// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotKeys basic 1`] = `
<div
  class="mainDiv"
>
  <div
    class="unit-test-input"
  />
  <div>
    <div
      class="keyhook"
    >
      обработчик keydown сработал: 
      1
    </div>
  </div>
</div>
`;

exports[`HotKeys комплексный тест - 2 пересоздающихся соседа: complex 1`] = `
<div
  class="mainDiv"
>
  <div
    class="unit-test-input"
  />
  <div>
    <div>
      <div
        class="keyhook"
      >
        обработчик keydown сработал: 
        1
      </div>
    </div>
  </div>
  <div>
    <div>
      <div
        class="keyhook"
      >
        обработчик keydown сработал: 
        0
      </div>
    </div>
  </div>
</div>
`;

exports[`HotKeys комплексный тест - первый сосед пропадает, второй должен перехватить инициативу: complex 1`] = `
<div
  class="mainDiv"
>
  <div
    class="unit-test-input"
  />
  <article />
  <div>
    <div>
      <div
        class="keyhook"
      >
        обработчик keydown сработал: 
        1
      </div>
    </div>
  </div>
</div>
`;

exports[`HotKeys комплексный тест - перерисовывающийся и пересоздающийся соседи: complex 1`] = `
<div
  class="mainDiv"
>
  <div
    class="unit-test-input"
  />
  <div
    class="abc2"
  >
    <div>
      <div
        class="keyhook"
      >
        обработчик keydown сработал: 
        1
      </div>
    </div>
  </div>
  <div>
    <div>
      <div
        class="keyhook"
      >
        обработчик keydown сработал: 
        0
      </div>
    </div>
  </div>
</div>
`;

exports[`HotKeys родительский контрол перерисовывается 1`] = `
<div
  class="mainDiv"
>
  <div
    class="unit-test-input"
  />
  <div
    class="abc"
  >
    <div>
      <div
        class="keyhook"
      >
        обработчик keydown сработал: 
        1
      </div>
    </div>
  </div>
</div>
`;

exports[`HotKeys родительский контрол пересоздается 1`] = `
<div
  class="mainDiv"
>
  <div
    class="unit-test-input"
  />
  <article>
    <div>
      <div
        class="keyhook"
      >
        обработчик keydown сработал: 
        1
      </div>
    </div>
  </article>
</div>
`;
